src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  actions.forEach(action => {
+    if (action.type === 'clear') {
+      currentState = {};
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      currentState = { ...currentState };
+      action.keysToRemove.forEach(key => {
+        delete currentState[key];
+      });
+    }
+    stateHistory.push(currentState);
```

Comment:
You are pushing the reference of currentState into the stateHistory array. This means all elements in stateHistory will reference the same object. To fix this, push a copy of currentState using the spread operator or Object.assign.